src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,75 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
-                if (idleGuard.Release()->try_push(std::make_pair(workerRequestPtr->_index, workerRequestPtr))) {
-                    // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                    // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                    IE::Task t;
-                    do {
-                        _inferPipelineTasks.try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t)));
-                    do {
-                        _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    if (workerRequestPtr->_exceptionPtr != nullptr) {
+                        if (workerRequestPtr->_testExec->count == 1) {
+                            auto capturedTask = std::move(workerRequestPtr->_task);
+                            capturedTask();
+                        } else {
+                            workerRequestPtr->_reload = true;
+                            auto capturedTask = std::move(workerRequestPtr->_testExec->_task);
+                            workerRequestPtr->_testExec->count++;
+                            selectOtherDevice(workerRequestPtr->_deviceName);
+                            if (_loadContext[FALLBACKDEVICE].isAlready) {
+                                workerRequestPtr->_startTimes.pop_back();
+                            }
+                            capturedTask();
+                        }
+                    } else {
+                        workerRequestPtr->_testExec->count = 0;
+                        workerRequestPtr->_reload = false;
+                        auto capturedTask = std::move(workerRequestPtr->_task);
+                        capturedTask();
+                    }
+                    // try to return the request to the idle list (fails if the overall object destruction has began)
+                    if (idleGuard.Release()->try_push(std::make_pair(workerRequestPtr->_index, workerRequestPtr))) {
+                        // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
+                        // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
+                        IE::Task t;
+                        do {
+                            _inferPipelineTasks.try_pop(t);
+                        } while (t && ScheduleToWorkerInferRequest(std::move(t)));
+                        do {
+                            _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
+                        } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    }
                 }
             });
     }
 }
 
+void AutoSchedule::selectOtherDevice(std::string currentDeviceName) {
+    std::unique_lock<std::mutex> lck(_mutex);
```

Comment:
we should implement this to be called only once, like if many infer requests fails, we don;t need to load fallback device for several times.

Reply:
If re-infer and failed again, I think it need to retry another devices again, so just add some lock here but it can be called in many times.

Reply:
I mean this situation, say if we have infer_request_1_gpu and infer_request_2_gpu both failed in separate threads, no matter which one triggered the selectOtherDevice, we only expect this select to be called only once, and the other one just need to re-trigger the inference. (all these operations need to be thread-safe)

Reply:
also, some other points may need consider, say we have fallbackdevice successfully enabled, shall we release the resources of actualdevice?

Reply:
Yes, Added call_once func to release ACTUALDEVICE when FALLBACKDEVICE is ready.